### ISSUE TITLE
Export link locations change upon modifications to exported content

### ISSUE THREAD

**Issue by lgeistlinger** [waldronlab/BugSigDB#94]
The "16s variable region" field doesn't seem to export properly lately as it comes out all blank in the [experiment data export](https://tinyurl.com/yb2fmpa3).

**Comment by lgeistlinger** [comment]
The problem seems to be that the underlying link has changed from

BEFORE: 

"https://bugsigdb.org/w/index.php?title=Special:Ask&x=-5B-5BCategory%3AExperiments-5D-5D%2F-3FOriginal-20page-20name%3DExperiment-20page-20name%2F-3FRelated-20study%3DStudy%2F-3FLocation-20of-20subjects%2F-3FHost-20species%2F-3FBody-20site%2F-3FCondition%2F-3FGroup-200-20name%2F-3FGroup-201-20name%2F-3FGroup-201-20definition%2F-3FGroup-200-20sample-20size%2F-3FGroup-201-20sample-20size%2F-3FAntibiotics-20exclusion%2F-3FSequencing-20type%2F-3F16s-20variable-20region%2F-3FSequencing-20platform%2F-3FStatistical-20test%2F-3FSignificance-20threshold%2F-3FMHT-20correction%2F-3FLDA-20Score-20above%2F-3FMatched-20on%2F-3FConfounders-20controlled-20for%2F-3FPielou%2F-3FShannon%2F-3FChao1%2F-3FSimpson%2F-3FInverse-20Simpson%2F-3FRichness&mainlabel=-&limit=5000&order=asc&sort=Page%20sort%20number&offset=0&format=csv&searchlabel=Download%20all%20Experiments%20%28CSV%29&filename=experiments.csv"

NOW:

"https://bugsigdb.org/w/index.php?title=Special:Ask&x=-5B-5BCategory%3AExperiments-5D-5D%2F-3FOriginal-20page-20name%3DExperiment-20page-20name%2F-3FRelated-20study%3DStudy%2F-3FLocation-20of-20subjects%2F-3FHost-20species%2F-3FBody-20site%2F-3FUBERON-20ID%2F-3FCondition%2F-3FEFO-20ID%2F-3FGroup-200-20name%2F-3FGroup-201-20name%2F-3FGroup-201-20definition%2F-3FGroup-200-20sample-20size%2F-3FGroup-201-20sample-20size%2F-3FAntibiotics-20exclusion%2F-3FSequencing-20type%2F-3F16S-20variable-20region%2F-3FSequencing-20platform%2F-3FStatistical-20test%2F-3FSignificance-20threshold%2F-3FMHT-20correction%2F-3FLDA-20Score-20above%2F-3FMatched-20on%2F-3FConfounders-20controlled-20for%2F-3FPielou%2F-3FShannon%2F-3FChao1%2F-3FSimpson%2F-3FInverse-20Simpson%2F-3FRichness%2F-3FState%2F-3FReviewer&mainlabel=-&limit=5000&order=asc&sort=Page%20sort%20number&offset=0&format=csv&searchlabel=Download%20all%20Experiments%20%28CSV%29&filename=experiments.csv"

However, we've written [code / an R package](https://github.com/waldronlab/bugsigdbr) that pulls the export files from a specified link. This code still pulls the link from "BEFORE" as it assumes this to be the stable link for bulk export of here experiment data. These links however seem to undergo changes as also observed for https://github.com/waldronlab/BugSigDB/issues/92. 

This brings us back to that we need simplified and stable links for obtaining the bulk export files as mentioned before. https://github.com/waldronlab/BugSigDB/issues/16#issuecomment-731292257

**Comment by tosfos** [comment]
Ah, OK. That makes sense. I don't think there is a way to provide a stable link though, from the MediaWiki side. Can you modify the script to scrape the current URL from the Help:Export page? We can separate the 3 links to be stored on separate pages if that would help. (We can have one of our developers modify the R code if needed.)

If that is not possible, we can store a static CSV on the server at a set URL and write a script to occasionally (probably overnight) update this CSV with the latest data. This would also speed up the CSV download.

**Comment by lgeistlinger** [comment]
It sounds like the second option would make for a cleaner self-contained solution directly on bugsigdb.org, but I am also ok with the scraping for our R package. 
This will be fine as long as we are using these export links only internally, and as an intermediate prior to upload a stable release under a specified link on zenodo - which is then the entry point for external applications. 

But I think we will then need a note on the help:export page documenting this behavior for other developers to not run into the same issue, incl pointing them to the zenodo for a stable link, and our R package for programmatic access. I could take care of that.

**Comment by lwaldron** [comment]
I don't feel strongly but the scraping alternative has some advantages:
1) it means that the bulk download from bugsigdb.org is current up to the minute (or hour?)
2) the plan is for daily dumps to be available anyways at https://github.com/waldronlab/bugsigdbexports, plus periodic releases with DOIs on Zenodo, and we can point people to those places for stable links

**Comment by lwaldron** [comment]
Come to think of it @lgeistlinger , why don't we use the scraping solution for the GitHub Action at https://github.com/waldronlab/BugSigDBExports, but then use BugSigDBExports as the data source for bugsigdbr? We can easily increase the update frequency there to daily. It would shift load from our server to github's servers, and with commit hash or version as an argument it would make reproducible analysis easy.

**Comment by lgeistlinger** [comment]
> but then use BugSigDBExports as the data source for bugsigdbr?

It's a good idea. Currently, there is some circular logic here as `BugSigDBExports` uses `bugsigdbr` to pull the data from bugsigdb.org (via `bugsigdbr`'s `dump_release.R` script).
But if we wanted to resolve that, then

1. the scraping solution + the current body of `bugsigdbr::importBugSigDB` would need to go into `bugsigdbr`'s `dump_release.R` script.
2. `bugsigdbr::importBugSigDB` would need to be rewritten to pull from `BugSigDBExports`.

Not too much work, and a nice side effect of it would be that we would relieve `bugsigdbr` from depending on functionality for scraping such as `rvest`.

**Comment by lgeistlinger** [comment]
@jwokaty and I have reworked `bugsigdbr` and `BugSigDBExports` [as specified](https://github.com/waldronlab/BugSigDB/issues/94#issuecomment-903121949). As of `bugsigdbr_0.99.7`, `bugsigdbr::importBugSigDB` pulls the data from `BugSigDBExports`, which in turn pulls and assembles the data from bugsigdb.org. Closing here, follow-up can be done via raising an issue under `bugsigdbr` or `BugSigDBExports`.
